test/refactoring/ConvertAsync/convert_async_wrapper.swift

```diff
@@ -7,7 +7,7 @@ enum CustomError : Error {
 // RUN: %refactor-check-compiles -add-async-wrapper -dump-text -source-filename %s -pos=%(line+1):1 -enable-experimental-concurrency | %FileCheck -check-prefix=FOO1 %s
```

Comment:
Could you add an async wrapper test just to make sure we properly handle a forwarding call to a completion handler param that isn't the last parameter?

Reply:
Good catch 👍

Reply:
Added `handler_position.swift` which tests the handler at the start + middle. Both the function declaration added and the forwarding call were broken - fixed now.

Also added some comments before/after each arg which caught some more issues. Fixed those too.